### PR TITLE
Random username generation for the anonymous user

### PR DESCRIPTION
**Pull request by ajainuary**
This replaces the 'Anonymous Visitor' greeting in the Wiki's Top Navbar to the user's randomly computer generated human friendly english username composed of 2 words.

**How to test?**
`git pull`
`git checkout random-username`
Since it introduces a new package, please do
`npm start yarn`
`npm start server`
View the wiki as an Anonymous user (i.e. without login)

### CODE REVIEW

**Comment by houshuang** [comment]
Note that you don't need to do git clean, etc. Simply do npm start yarn, and it will update all the packages.

File: frog/imports/client/Wiki/components/TopNavbar/PrimaryButton.js [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 
 import * as React from 'react';
 
 type PrimaryButtonPropsT = {
   title: string,
   active?: boolean,
   icon?: React.ComponentType<*>,
-  callback?: () => void
+  callback?: () => void,
+  i?: boolean
```
Comment (houshuang) [inline frog/imports/client/Wiki/components/TopNavbar/PrimaryButton.js]: this variable is not very descriptive, maybe rename as italic

File: frog/imports/client/Wiki/components/TopNavbar/index.js [hunk 2]
```diff
@@ -7,8 +7,9 @@ import OverflowPanel from './OverflowPanel';
 
 type TopNavBarPropsT = {
   /** The current Meteor username */
-  user: string,
-
+  username: string,
+  /* Status of the user, username will be displayed in italics if true */
```
Comment (houshuang) [inline frog/imports/client/Wiki/components/TopNavbar/index.js]: we prefer // comments unless they are multiline

File: frog/imports/client/Wiki/components/TopNavbar/index.js [hunk 3]
```diff
@@ -7,8 +7,9 @@ import OverflowPanel from './OverflowPanel';
 
 type TopNavBarPropsT = {
   /** The current Meteor username */
```
Comment (houshuang) [inline frog/imports/client/Wiki/components/TopNavbar/index.js]: this kind of comment is not very useful. Try to limit comments to things that are not obvious.

File: frog/imports/client/Wiki/index.js [hunk 4]
```diff
@@ -110,7 +111,8 @@ class WikiComp extends Component<WikiCompPropsT, WikiCompStateT> {
       deletedPageModalOpen: false,
       currentDeletedPageId: null,
       currentDeletedPageTitle: null,
-      rightSideCurrentPageObj: null
+      rightSideCurrentPageObj: null,
+      user: Meteor.user()
```
Comment (houshuang) [inline frog/imports/client/Wiki/index.js]: why would you put this in state? Why can't you just query Meteor.user() when you need it? It might make sense if you were actually listening to the Meteor.user and updating state when it changes, but you don't (and we don't expect it to change, except if we allowed renaming in the future, as discussed on Slack)

File: frog/imports/client/Wiki/index.js [hunk 5]
```diff
@@ -223,7 +225,7 @@ class WikiComp extends Component<WikiCompPropsT, WikiCompStateT> {
     if (!currentPageObj) {
       if (!fullPageObj.noNewInstances) {
         this.initialLoad = true;
-        const instanceName = Meteor.user().username;
+        const instanceName = this.state.user.username;
```
Comment (houshuang) [inline frog/imports/client/Wiki/index.js]: why this change?

File: frog/server/user_accounts.js [hunk 6]
```diff
@@ -13,7 +17,7 @@ const doLogin = (user, self) => {
       return Accounts._loginUser(self, alreadyUser._id);
     }
   }
-  const userServiceData = { id: user || uuid() };
+  const userServiceData = { id: user || uniqueNamesGenerator(usernameConfig) };
```
Comment (houshuang) [inline frog/server/user_accounts.js]: can we make the username capital first letters, like Happy Panda instead of happy panda?

**Review by houshuang — CHANGES_REQUESTED** [review]

File: frog/imports/client/Wiki/index.js (hunk 4 above)
Comment (ajainuary) [reply frog/imports/client/Wiki/index.js]: On the server, this will fetch the record from the database. To improve the latency since it is being called multiple times, we could save the returned record to a variable instead of re-calling `Meteor.user()`.

File: frog/imports/client/Wiki/components/TopNavbar/index.js (hunk 3 above)
Comment (ajainuary) [reply frog/imports/client/Wiki/components/TopNavbar/index.js]: This comment is not a part of the PR.

File: frog/server/user_accounts.js (hunk 6 above)
Comment (ajainuary) [reply frog/server/user_accounts.js]: Made the changes

File: frog/imports/client/Wiki/components/TopNavbar/index.js (hunk 3 above)
Comment (ajainuary) [reply frog/imports/client/Wiki/components/TopNavbar/index.js]: Removed the comment

File: frog/imports/client/Wiki/index.js (hunk 5 above)
Comment (ajainuary) [reply frog/imports/client/Wiki/index.js]: In order to improve the performance, we re-use the value instead of calling `Meteor.user()` again.

File: frog/imports/client/Wiki/components/TopNavbar/index.js (hunk 2 above)
Comment (ajainuary) [reply frog/imports/client/Wiki/components/TopNavbar/index.js]: Made the changes

File: frog/imports/client/Wiki/components/TopNavbar/PrimaryButton.js (hunk 1 above)
Comment (ajainuary) [reply frog/imports/client/Wiki/components/TopNavbar/PrimaryButton.js]: Made the changes

File: frog/imports/client/Wiki/index.js (hunk 4 above)
Comment (houshuang) [reply frog/imports/client/Wiki/index.js]: That's not how Meteor works - Meteor uses something called Minimongo, which subscribes to certain collections and always keeps them in sync locally. The only time a server call is made is when you make a new subscription (with withTracker) or when you call a Meteor method. Please revert this.

**Review by houshuang — CHANGES_REQUESTED** [review]

**Comment by houshuang** [comment]
Note that you don't need to do git clean, etc. Simply do npm start yarn, and it will update all the packages.

File: frog/imports/client/Wiki/index.js (hunk 4 above)
Comment (ajainuary) [reply frog/imports/client/Wiki/index.js]: Reverted the changes

**Comment by ajainuary** [comment]
Reverted to using `meteor.user()`

**Review by houshuang — APPROVED** [review]
